managers/memory/file_ops.c

```diff
@@ -49,11 +49,15 @@ struct lego_file *file_open(struct lego_task_struct *tsk, const char *filename)
 	strncpy(file->filename, filename, MAX_FILENAME_LEN);
 
 #ifdef CONFIG_USE_RAMFS
-	file->f_op = &ramfs_file_ops;
+	unsigned int node0 = 0;
+	if(tsk->node==node0)
+		file->f_op = &ramfs_file_ops;
+	else
+		file->f_op = &ramfs_file_ops2;
```

Comment:
this needs to be changed. btw, should msg be a file data structure? this is Linux way of treating everything as a file. We don't have to do that and can/should create a separate data structure for messages.